util/dvsim/SynCfg.py

```diff
@@ -343,6 +343,50 @@ def _create_entry(val, norm=1.0, total=None, perctag="%"):
             else:
                 results_str += "No power report found\n\n"
 
+            # Append detailed messages if they exist
```

Comment:
This looks sensible enough, but I think there might be an easier way to do the same thing. If you define a list of pairs `[("Flow Errors", "flow_errors"), ("Flow Warnings", "flow_warnings"), ...]`, then you can iterate over that list and avoid having to repeat quite so much. This would also allow you to avoid the gigantic sum expression in the test. You would have something like

```python
has_msg = False
for _, key in my_pairs:
    if key in self.result['messages']:
        has_msg = True
        break
if has_msg and not self.args.publish:
    results_str += "\n### Errors and Warnings for Build Mode `'" + mode.name + "'`\n"
    for hdr, key in my_pairs:
        msgs = self.result['messages'].get(key)
        if msgs is not None:
            results_str += print_msg_list('#### ' + hdr, msgs, self.max_msg_count)
```

Reply:
thanks, this is a good suggestion! I'll modify both the synthesis and lint classes accordingly.